actionpack/lib/action_dispatch/http/parameters.rb

```diff
@@ -54,7 +54,7 @@ def parameters
                  rescue EOFError
                    query_parameters.dup
                  end
-        params.merge!(path_parameters)
+        params.merge!(path_parameters_utf8_enforced)
```

Comment:
This will now force the path params to be UTF8. and `set_binary_encoding` will force some parameters to be Binary.
I wonder if after those two methods we should validate that the parameters have valid encoding set to them.
Otherwise, this could cause only errors when the caller of the parameter try to use it.